scene/gui/code_edit.cpp

```diff
@@ -1763,6 +1815,89 @@ TypedArray<int> CodeEdit::get_folded_lines() const {
 	return folded_lines;
 }
 
+/* Code region */
+void CodeEdit::create_code_region() {
+	// Abort if there is no selected text
+	if (!has_selection()) {
+		return;
+	}
+	// Check that region tag find a comment delimiter and is valid
+	if (code_region_start_string.is_empty()) {
+		WARN_PRINT_ONCE("Cannot create code region without any one line comment delimiters");
+		return;
+	}
+	begin_complex_operation();
+	// Merge selections if selection starts on the same line the previous one ends
+	Vector<int> caret_edit_order = get_caret_index_edit_order();
+	for (int i = 1; i < caret_edit_order.size(); i++) {
+		int current_caret = caret_edit_order[i - 1];
+		int next_caret = caret_edit_order[i];
+		if (get_selection_from_line(current_caret) == get_selection_to_line(next_caret)) {
+			select(get_selection_from_line(next_caret), get_selection_from_column(next_caret), get_selection_to_line(current_caret), get_selection_to_column(current_caret), next_caret);
+			remove_caret(current_caret);
```

Comment:
Method looks good. `remove_caret` will change the remaining `caret` indexes, should update `caret_edit_order` or remove them in a separate loop.

Reply:
Carets are now deleted in a separate loop.